nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -581,17 +581,19 @@ sub multiLineList {
 EOF
 
 sub generateNetworkingDhcpConfig {
+    # FIXME disable networking.useDHCP by default when switching to networkd.
     my $config = <<EOF;
-  # The global useDHCP flag is deprecated, therefore explicitly set to false here.
-  # Per-interface useDHCP will be mandatory in the future, so this generated config
-  # replicates the default behaviour.
-  networking.useDHCP = lib.mkDefault false;
+  # Enables DHCP on each ethernet and wireless interface. In case of scripted networking
+  # (the default) this is the recommended approach. When using systemd-networkd it's
+  # still possible to use this option, but it's recommended to explicitly set it
+  # per-interface using `networking.interfaces.<interface>.useDHCP`.
```

Comment:
I'd say that the optimal configuration is to have _both_ when using networkd: explicit useDHCP per interface for the ones that should be configured for `network-online.target`, because they're the ones usually used for internet access, _and_ global useDHCP for any extra interfaces that are used occasionally.